### ISSUE TITLE
DeepLabV3: segment only person

### ISSUE THREAD

**Issue by muna-cs** [pytorch/vision#1520]
How can I segment person only and skip the other classes by using DeepLabV3?

**Comment by fmassa** [comment]
Just take the full output from `DeepLabV3`, apply a softmax over the classes, and then remove the classes you are not interested in.

```python
model = torchvision.models.segmentation.deeplabv3_resnet101(pretrained=True)
model.eval()
input = torch.rand(1, 3, 300, 300)

res = model(input)
out = res['out']
scores = torch.nn.functional.softmax(out, dim=1)

# person class is index 15, from https://pytorch.org/docs/stable/torchvision/models.html#semantic-segmentation
person = scores[:, 15, :, :]
```

**Comment by muna-cs** [comment]
I get a black image as output. I do not know where is the mistake?
This is my code:

```
preprocess = transforms.Compose([
        transforms.ToTensor(),
        transforms.Normalize(mean=[0.485, 0.456, 0.406], std=[0.229, 0.224, 0.225]),
 ])

model = torchvision.models.segmentation.deeplabv3_resnet101(pretrained=True)
model.eval()
input_img2=Image.open(filename)
input_tensor = preprocess((input_img2))
input_batch = input_tensor.unsqueeze(0)
if torch.cuda.is_available():
            input_batch = input_batch.to('cuda')
            model.to('cuda')
with torch.no_grad():
            output = model(input_batch)['out']
scores = torch.nn.functional.softmax(output, dim=1)
output_predictions = scores[ : , 15, : , : ].argmax(0)
palette = torch.tensor([2 ** 25 - 1, 2 ** 15 - 1, 2 ** 21 - 1])
colors = torch.as_tensor([i for i in range(21)])[:, None] * (palette)
colors = (colors % 512).numpy().astype("uint8")
 r = Image.fromarray(output_predictions.byte().cpu().numpy()).resize(input_img2.size)
 r.putpalette(colors)
 f, ax = plt.subplots(1, 2)
 ax[0].set_title('input image')
 ax[0].axis('off')
 ax[0].imshow(input_img2)
 ax[1].set_title('output')
 ax[1].axis('off')
 ax[1].imshow(r)
```

**Comment by fmassa** [comment]
@muna-cs your problem is when computing the argmax.

You should just do
```
scores = torch.nn.functional.softmax(output, dim=1)
# you can do this
output_predictions = scores[ : , : , : ].argmax(1) == 15
# or
output_predictions = scores[ : , 15, : , : ] > 0.5 # or any threshold you want


```

**Comment by muna-cs** [comment]
it is working well now
Thank you
